### DISCUSSION TITLE
如何让编辑器以及右侧栏跟着后缀走？

### DISCUSSION THREAD

**Discussion by hnhhzy** [opensumi/core#2371] [Q&A]
编辑器的话，我通过实现ClientAppContribution, BrowserEditorContribution, CommandContribution参考了[use-antd](https://github.com/opensumi/opensumi-module-samples/tree/main/modules/use-antd)代码实现了后缀关联编辑器。
看到新增的[app-desktop](https://github.com/opensumi/app-desktop)实现了侧边栏。

可以将他们两关联起来吗？特定的后缀才会打开这个侧边栏。其他的情况下是不显示的，还是这个侧边栏需要在editor里自己实现？

**Comment by erha19** [comment]
@hnhhzy 需要你自行实现这种逻辑，可以在判断到后缀文件打开时，去调用面板 API 展开侧边栏，参考案例 https://github.com/opensumi/opensumi-module-samples/blob/main/modules/toggle-panel/browser/toggle-panel.contribution.ts

**Comment by hnhhzy** [comment]
另外，新出的这个app-desktop和ide-electron有啥区别吗？我看结构好像不太一样，app-desktop有一个base部分

**Reply by erha19** [reply]
App Desktop 示例是一个类 ”小程序开发工具“ 的启动示例，提供了前置页、布局定制等更多能力的定制示例

**Comment by hnhhzy** [comment]
有创建一个右边栏的完整demo吗？ 并且可以传递参数的

**Reply by erha19** [reply]
@hnhhzy 可以参考一下这个案例 https://github.com/opensumi/opensumi-module-samples/blob/main/modules/custom-view/README.md ，同时这类自定义视图内有完整的内置服务能力，你可以通过事件、命令等机制去实现参数的传递。
